aderyn_core/src/report/markdown_printer.rs

```diff
@@ -0,0 +1,262 @@
+use crate::context::loader::ContextLoader;
+use std::io::{Result, Write};
+
+use super::{
+    printer::ReportPrinter,
+    reporter::{Issue, Report},
+};
+
+pub struct MarkdownReportPrinter;
+
+impl ReportPrinter<()> for MarkdownReportPrinter {
+    fn print_report<W: Write>(
```

Comment:
Could we print each of the issues in the JSON file, also?

As well as `issues_found` summary section, another section called `issues` that has members: `critical`, `high`,... etc. Then within each of those, an array of every issue found.

Reply:
In fact, it makes sense to produce a JSON output that mimicks the MD:
```json
{
  "files_summary": {...},
  "files_details": {...},
  "issue_summary": {...},
  "critical_issues": {...},
  "high_issues": {...},
  ...
}
```

Reply:
Done 👊

Reply:
Oh wait..there's 1 more change i wanna make

Reply:
Done 👊